File: Campus_Services_Portal/Models/Entities/Event.cs
namespace Campus_Services_Portal.Models.Entities
{
    public class Event
    {
        public int Id { get; set; }

        public string Title { get; set; } = string.Empty;

        public string Description { get; set; } = string.Empty;

        public DateTime EventDate { get; set; }

        public string Venue { get; set; } = string.Empty;

        public int Capacity { get; set; }

        public bool IsActive { get; set; } = true;

        // One Event can have many registrations
        public ICollection<EventRegistration> EventRegistrations { get; set; }
            = new List<EventRegistration>();
    }
}

File: campusx-frontend/src/app/core/models/event.ts
// =====================================================
// EVENT
// =====================================================

export interface Event {
  id: number;
  title: string;
  description: string;
  eventDate: string;
  venue: string;
  capacity: number;
  isActive: boolean;
}


// =====================================================
// CREATE EVENT
// =====================================================

export interface CreateEvent {
  title: string;
  description: string;
  eventDate: string;
  venue: string;
  capacity: number;
}


// =====================================================
// UPDATE EVENT
// =====================================================

export interface UpdateEvent {
  title: string;
  description: string;
  eventDate: string;
  venue: string;
  capacity: number;
  isActive: boolean;
}


// =====================================================
// EVENT REGISTRATION
// =====================================================

export interface EventRegistration {
  id: number;
  eventId: number;
  eventTitle?: string;
  studentId: number;
  registeredAt: string;
}


// =====================================================
// CREATE EVENT REGISTRATION
// =====================================================

export interface CreateEventRegistration {
  eventId: number;
}